src/main/java/dal/fatih/todoly/HibernateTaskRepository.java

```diff
@@ -0,0 +1,112 @@
+package dal.fatih.todoly;
+
+
+import javax.persistence.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.persistence.metamodel.EntityType;
+import java.sql.Date;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+
+public class HibernateTaskRepository implements TaskRepository {
+
+    private final EntityManagerFactory entityManagerFactory
+            = Persistence.createEntityManagerFactory("Todoly");
+    private EntityManager entityManager = entityManagerFactory.createEntityManager();
+    private CriteriaBuilder cb;
+    private CriteriaQuery<Task> cq;
+    private Root<Task> taskRoot;

```

Comment:
Never create variables as a class field and keep changing it. Always re-create them. Also, try to define all the class fields as `final` and initialize them in the constructor if possible. 

Simply remove these lines.